parser/parser.go

```diff
@@ -306,19 +306,8 @@ func (p *parser) parseSecondary() Node {
 	case Number:
 		p.next()
 		value := strings.Replace(token.Value, "_", "", -1)
-		if strings.Contains(value, "x") {
-			number, err := strconv.ParseInt(value, 0, 64)
-			if err != nil {
-				p.error("invalid hex literal: %v", err)
-			}
-			if number > math.MaxInt {
-				p.error("integer literal is too large")
-				return nil
-			}
-			node := &IntegerNode{Value: int(number)}
-			node.SetLocation(token.Location)
-			return node
-		} else if strings.ContainsAny(value, ".eE") {
+		isHex := strings.HasPrefix(value, "0x") || strings.HasPrefix(value, "0X")
+		if !isHex && strings.ContainsAny(value, ".eE") {
```

Comment:
We are checking here only for `!isHex`. What about others?

Reply:
We are specifically excluding hexadecimal literals containing the letter 'E' when performing float conversions.